### PR TITLE
Clean up some DI code

### PR DESCRIPTION
**Pull request by davidfowl**
- Remove empty CreateInstanceCallSite file
- Removed ServiceScopeFactoryCallsite and used ConstantCallsite instead.
- Added fast path if the value is already cached on the callsite.

### CODE REVIEW

**Review by maryamariyan — APPROVED** [review]

File: src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteValidator.cs [hunk 1]
```diff
@@ -78,7 +78,7 @@ protected override Type VisitRootCache(ServiceCallSite singletonCallSite, CallSi
         protected override Type VisitScopeCache(ServiceCallSite scopedCallSite, CallSiteValidatorState state)
         {
             // We are fine with having ServiceScopeService requested by singletons
-            if (scopedCallSite is ServiceScopeFactoryCallSite)
```
Comment (pakrym) [inline src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteValidator.cs:81]: Can you replace ServiceScopeFactoryCallSite with a constant call site?

Comment (pakrym) [reply src/libraries/Microsoft.Extensions.DependencyInjection/src/ServiceLookup/CallSiteValidator.cs:81]: Hah. You did.

**Review by pakrym — APPROVED** [review]

**Comment by davidfowl** [comment]
Test failures are unrelated. SocketsHttpHandler test timing out, mono interpreter test with System.Security.Cryptography.Xml.Tests are crashing.

**Comment by kunalspathak** [comment]
Perf regression in arm64: https://github.com/DrewScoggins/performance-2/issues/7130

**Comment by kunalspathak** [comment]
Perf improvements in arm64: https://github.com/DrewScoggins/performance-2/issues/7137

**Comment by davidfowl** [comment]
So it both regressed and improved performance. I like it.

**Comment by kunalspathak** [comment]
Are the regressions expected?

**Comment by davidfowl** [comment]
No, I'm not sure why it regressed.

**Comment by davidfowl** [comment]
My guess is that this change https://github.com/dotnet/runtime/pull/55340 fixed the regression.

**Comment by kunalspathak** [comment]
Looks to be the case:
image

**Comment by davidfowl** [comment]
Excellent

**Pull request merged**